### PR TITLE
React testing

### PR DESCRIPTION
**Pull request by Ntwali-Josue**
### In this Pull Request, We: 

- Write unit tests for the files operate.js and calculate.js using Jest.
- Create unit tests for all major components.

### CODE REVIEW

File: src/tests/Calculator.test.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+beforeEach(() => {
+  render(<Calculator />);
+});
+
+describe('Test Calculator Buttons', () => {
+  test('Button 7 to be in the document', () => {
+    const elementSeven = screen.getByTestId('seven');
+    expect(elementSeven).toBeInTheDocument();
+  });
```
Comment (henatan99) [inline src/tests/Calculator.test.js:13]: - [x] Very nice work here. I think it is great to add snapshot tests for each component as it is a simple yet powerful test. :pray: 
- [x] You can also add some tests for some user interaction. For example, when the `AC` button is clicked the result text should change to 0. Kindly make at least one test for user interaction. :pray:

File: src/tests/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      total: null,
+      next: null,
+      operation: null,
+    };
+    obj = calculate(obj, '8');
+    obj = calculate(obj, 'x');
+    obj = calculate(obj, '2');
+    obj = calculate(obj, '=');
+    expect(obj.total).toEqual('16');
+  });
```
Comment (henatan99) [inline src/tests/calculate.test.js:28]: - [x] Amazing work so far. But all the scenarios need to be tested similarly as per the project's requirement. Kindly add tests for the other operations too. :pray:

File: src/tests/operate.test.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    const firstNumber = 1;
+    const secondNumber = 2;
+    const operationMinus = '-';
+    expect(operate(firstNumber, secondNumber, operationMinus)).toBe('-1');
+  });
+  test('Seven ( 7 ) button exists', () => {
+    const sevenButton = screen.getByTestId('seven');
+    expect(sevenButton).toBeInTheDocument();
+  });
+});
```
Comment (henatan99) [inline src/tests/operate.test.js:35]: - [x] Same goes here. You did a great job having the tests for some of the logic, but it is needed to test all the scenarios as per the project's requirement. :pray:

File: src/components/Calculator.js [hunk 4]
```diff
@@ -39,12 +38,19 @@ const Calculator = () => {
   return (
```
Comment (henatan99) [inline src/components/Calculator.js:38]: - [ ] When an operator button is clicked multiple times consecutively, it throws an error on the browser. Kindly fix this error or implement an error handling function or condition. Your app needs to be free of uncaught errors. :pray: 

Screenshot from 2021-09-24 02-39-54

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @Ntwali-Josue , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by barackm — APPROVED** [review]
Hi @Ntwali-Josue and @zainsadaqat, 👋 

# STATUS: Approved  🎉 
Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by zainsadaqat** [comment]
Thanks, @barackm for approving our pull request!

**Pull request merged**